execution/executor/src/db_bootstrapper.rs

```diff
@@ -22,17 +28,29 @@ pub fn maybe_bootstrap_db<V: VMExecutor>(db: DbReaderWriter, config: &NodeConfig
         .execution
         .genesis
         .as_ref()
-        .expect("failed to load genesis transaction!")
+        .expect("failed to load genesis txn!")
         .clone();
 
+    execute_and_commit::<V>(db, genesis_txn).map(|_| ())
+}
+
+pub fn compute_genesis_waypoint<V: VMExecutor>(txn: Transaction) -> Result<Waypoint> {
+    let path = TempPath::new();
+    path.create_as_dir().expect("Unable to create directory");
+    let (_, db) = DbReaderWriter::wrap(LibraDB::new(&path.path()));
+    let ledger_info = execute_and_commit::<V>(db, txn)?;
```

Comment:
aren't you going to execute twice by doing this here in config?

Reply:
Twice in what way? I put the block on the tests so this would only execute once...